config/samples/backends/edge/edge.yaml

```diff
@@ -0,0 +1,83 @@
+# Sample using Ceph as a glance backend
+# Requires a running Ceph cluster and its `/etc/ceph` files in secret `ceph-conf-files`
+# This can be achieved with the `ceph` target of `install_yamls`
+apiVersion: core.openstack.org/v1beta1
+kind: OpenStackControlPlane
+metadata:
+  name: openstack
+spec:
+  glance:
+    template:
+      databaseInstance: openstack
+      databaseAccount: glance
+      secret: osp-secret
+      storageClass: ""
+      storageRequest: 10G
+      keystoneEndpoint: az0
+      glanceAPIs:
+        az0:
+          type: split
+          replicas: 3
+          customServiceConfig: |
```

Comment:
```
          customServiceConfig: |
            [DEFAULT]
            [glance_store]
            stores=http,rbd
            os_region_name=regionOne
            default_backend = az0

            [az0]
            store_description = "az0 RBD backend"
            rbd_store_pool = images
            rbd_store_user = openstack
            rbd_store_ceph_conf = /etc/ceph/az0.conf

            [az1]
            store_description = "az1 RBD backend"
            rbd_store_pool = images
            rbd_store_user = az1.openstack
            rbd_store_ceph_conf = /etc/ceph/az1.conf

            [az2]
            store_description = "az2 RBD backend"
            rbd_store_pool = images
            rbd_store_user = az2.openstack
            rbd_store_ceph_conf = /etc/ceph/az2.conf
```

Reply:
I'm going to test this in my env . Maybe we can do that before we merge so we know we have a working CR?

Reply:
Sure, also, if you think there are more things to update, feel free to push within this branch!

Reply:
I verified the following [1] in my environment. You seem to have the same thing except the rbd_store_user in your example needs an update. From `az1.openstack` to just `openstack`. This works if your ceph configuration file has a client section defining the keyring [2].

[1]
```
  glance:
    apiOverrides:
      default:
        route: {}
    template:
      databaseAccount: glance
      databaseInstance: openstack
      glanceAPIs:
        az1:
          customServiceConfig: '[DEFAULT]

            enabled_import_methods = [web-download,copy-image,glance-direct]

            enabled_backends = az0:rbd,az1:rbd


            [glance_store]

            stores = http,rbd

            default_backend = az1


            [az1]

            rbd_store_ceph_conf = /etc/ceph/az1.conf

            store_description = "az1 RBD backend"

            rbd_store_pool = images

            rbd_store_user = openstack


            [az0]

            rbd_store_ceph_conf = /etc/ceph/ceph.conf

            store_description = "az0 RBD backend"

            rbd_store_pool = images

            rbd_store_user = openstack


            '
          networkAttachments:
          - storage
          override:
            service:
              internal:
                metadata:
                  annotations:
                    metallb.universe.tf/address-pool: internalapi
                    metallb.universe.tf/allow-shared-ip: internalapi
                    metallb.universe.tf/loadBalancerIPs: 172.17.0.81
                spec:
                  type: LoadBalancer
          preserveJobs: false
          replicas: 3
          type: edge
        az2:
          customServiceConfig: '[DEFAULT]

            enabled_import_methods = [web-download,copy-image,glance-direct]

            enabled_backends = az0:rbd,az2:rbd


            [glance_store]

            stores = http,rbd

            default_backend = az2


            [az2]

            rbd_store_ceph_conf = /etc/ceph/az2.conf

            store_description = "az2 RBD backend"

            rbd_store_pool = images

            rbd_store_user = openstack


            [az0]

            rbd_store_ceph_conf = /etc/ceph/az2.conf

            store_description = "az0 RBD backend"

            rbd_store_pool = images

            rbd_store_user = openstack

            enabled_import_methods = [web-download,copy-image,glance-direct]


            '
          networkAttachments:
          - storage
          override:
            service:
              internal:
                metadata:
                  annotations:
                    metallb.universe.tf/address-pool: internalapi
                    metallb.universe.tf/allow-shared-ip: internalapi
                    metallb.universe.tf/loadBalancerIPs: 172.17.0.82
                spec:
                  type: LoadBalancer
          preserveJobs: false
          replicas: 3
          type: edge
        default:
          customServiceConfig: '[DEFAULT]

            enabled_import_methods = [web-download,copy-image,glance-direct]

            enabled_backends = az0:rbd,az1:rbd,az2:rbd


            [glance_store]

            stores = http,rbd

            default_backend = az0


            [az0]

            rbd_store_ceph_conf = /etc/ceph/ceph.conf

            store_description = "az0 RBD backend"

            rbd_store_pool = images

            rbd_store_user = openstack


            [az1]

            rbd_store_ceph_conf = /etc/ceph/az1.conf

            store_description = "az1 RBD backend"

            rbd_store_pool = images

            rbd_store_user = openstack


            [az2]

            rbd_store_ceph_conf = /etc/ceph/az2.conf

            store_description = "az2 RBD backend"

            rbd_store_pool = images

            rbd_store_user = openstack


            '
          networkAttachments:
          - storage
          override:
            service:
              internal:
                metadata:
                  annotations:
                    metallb.universe.tf/address-pool: internalapi
                    metallb.universe.tf/allow-shared-ip: internalapi
                    metallb.universe.tf/loadBalancerIPs: 172.17.0.80
                spec:
                  type: LoadBalancer
          preserveJobs: false
          replicas: 3
          type: split
      keystoneEndpoint: default
      serviceUser: glance
      storageClass: local-storage
      storageRequest: 10G
```

[2]
```
[zuul@compute-7 ~]$ sudo cat /etc/ceph/{ceph,az1,az2}.conf | fgrep [client] -A 1
[client]
keyring = /etc/ceph/ceph.client.openstack.keyring
--
[client]
keyring = /etc/ceph/az1.client.openstack.keyring
--
[client]
keyring = /etc/ceph/az2.client.openstack.keyring
[zuul@compute-7 ~]$ 
```

Reply:
Also, I configure each edge site with two backends (the local ceph cluster and one remote ceph cluster -- usually the default one AKA central) so that images from an edge site can be copied back. I'll update the PR accordingly.